### PR TITLE
feat(rust-libp2p): update required jobs

### PR DESCRIPTION
**Pull request by thomaseizinger**
### Summary

Companion PR for https://github.com/libp2p/rust-libp2p/pull/4880.

### What else do we need to know?

**DRI:** myself

### Reviewer's Checklist

- [x] It is clear where the request is coming from (if unsure, ask)
- [x] All the automated checks passed
- [x] The YAML changes reflect the summary of the request
- [x] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

File: github/libp2p.yml
Comment (thomaseizinger) [inline github/libp2p.yml:7285]:
```suggestion
            - Compile libp2p in transport feature matrix
            - Compile libp2p-identity in feature matrix
```

File: github/libp2p.yml
Comment (thomaseizinger) [inline github/libp2p.yml:6508]: Not sure what this is doing in my PR?

Comment (mxinden) [reply github/libp2p.yml:6508]: @achingbrain @galargh do you why automation pushed these changes to an unrelated pull request?

Comment (galargh) [reply github/libp2p.yml]: The automation runs https://github.com/libp2p/github-mgmt/blob/master/scripts/src/actions/fix-yaml-config.ts after every push to the default branch and on every PR.

The reason why unrelated changes were pushed to the PR was that https://github.com/libp2p/github-mgmt/pull/197 has not been merged. The timeline would be, 1. the sync workflow discovers divergence between what's in this repo and the actual GitHub state, 2. the repo gets updated, 3. the PR with fixes is created but not merged, 4. this PR is created, 5. the fixes are also proposed in this PR because the PR with fixes was not merged yet.

Another thing to note is that once a fix PR is merged, then all the existing PRs are rebased which would have got rid of the unrelated changes.

It is a bit clunky at the moment. I'd like to package the workflows that do the actual work in this repo into nice reusable actions so it'd be easier for users to reason about and customise what happens when. There's no timeline for that work though.
 
Another issue, the commits with fixes are associated with PR owner at the moment which shouldn't be the case as they are bot commits. I created an issue for that here - https://github.com/ipdxco/github-as-code/issues/110

**Comment by mxinden** [comment]
@thomaseizinger do I understand correctly, that you want this merged? (https://github.com/libp2p/rust-libp2p/pull/4880#issuecomment-1915996090)

If so, can you confirm and set it to "Ready for review"?

**Review by p-shahi — APPROVED** [review]

**Comment by p-shahi** [comment]
lgtm @thomaseizinger

**Comment by thomaseizinger** [comment]
> lgtm @thomaseizinger

Thanks! The merge queue on rust-libp2p is currently stuck so please don't merge this yet. I'll ping again when it is ready.

**Comment by galargh** [comment]
Closing since it's been stale for a while. Please reopen if it's still needed.

**Pull request closed without merging**
